modules/auxiliary/crawler/msfcrawler.rb

```diff
@@ -327,68 +335,68 @@ def insertnewpath(hashreq)
   # Build a new hash for a local path
   #
 
-  def urltohash(m,url,basepath,dat)
+  def urltohash(method, url, basepath, dat)
+    # method: HTTP method
+    # url: uri?[query]
+    # basepath: base path/uri to determine absolute path when relative
+    # data: body data, nil if GET and query = uri.query
 
-      # m:   method
-      # url: uri?[query]
-      # basepath: base path/uri to determine absolute path when relative
-      # data: body data, nil if GET and query = uri.query
+    uri = URI.parse(url)
+    uritargetssl = (uri.scheme == 'https') ? true : false
 
-      uri = URI.parse(url)
-      uritargetssl = (uri.scheme == "https") ? true : false
+    uritargethost = uri.host
+    if uri.host.nil? || uri.host.empty?
+      uritargethost = ctarget
+      uritargetssl = cssl
+    end
 
-      uritargethost = uri.host
-      if (uri.host.nil? or uri.host.empty?)
-        uritargethost = self.ctarget
-        uritargetssl = self.cssl
-      end
+    uritargetport = uri.port
+    if uri.port.nil?
+      uritargetport = cport
+    end
 
-      uritargetport = uri.port
-      if (uri.port.nil?)
-        uritargetport = self.cport
-      end
+    uritargetpath = uri.path
+    if uri.path.nil? || uri.path.empty?
+      uritargetpath = '/'
+    end
 
-      uritargetpath = uri.path
-      if (uri.path.nil? or uri.path.empty?)
-        uritargetpath = "/"
+    newp = Pathname.new(uritargetpath)
+    oldp = Pathname.new(basepath)
+    if !newp.absolute?
+      if oldp.to_s[-1, 1] == '/'
+        newp = oldp + newp
+      elsif !newp.to_s.empty?
+        newp = File.join(oldp.dirname, newp)
       end
+    end
 
-      newp = Pathname.new(uritargetpath)
-      oldp = Pathname.new(basepath)
-      if !newp.absolute?
-        if oldp.to_s[-1,1] == '/'
-          newp = oldp+newp
-        else
-          if !newp.to_s.empty?
-            newp = File.join(oldp.dirname,newp)
-          end
-        end
-      end
+    hashreq = {
+      'rhost' => uritargethost,
+      'rport' => uritargetport,
+      'uri' => newp.to_s,
+      'method' => method,
+      'ctype' => 'text/plain',
+      'ssl' => uritargetssl,
+      'query' => uri.query,
+      'data' => nil
+    }
 
-      hashreq = {
-        'rhost'		=> uritargethost,
-        'rport'		=> uritargetport,
-        'uri' 		=> newp.to_s,
-        'method'   	=> m,
-        'ctype'		=> 'text/plain',
-        'ssl'		=> uritargetssl,
-        'query'		=> uri.query,
-        'data'		=> nil
-      }
-
-      if m == 'GET' and !dat.nil?
-        hashreq['query'] = dat
-      else
-        hashreq['data'] = dat
-      end
+    if (method == 'GET') && !dat.nil?
+      hashreq['query'] = dat
+    else
+      hashreq['data'] = dat
+    end
 
-      return hashreq
+    return hashreq
   end
 
-  # Taken from http://www.ruby-forum.com/topic/140101 by  Rob Biedenharn
+  # Abomonation taken from https://www.ruby-forum.com/t/creating-a-canonicalized-url/127036/6 by Rob Biedenharn
```

Comment:
It's spelt "Abomination" but don't think it's necessary to include here. 
```suggestion
  # Taken from https://www.ruby-forum.com/t/creating-a-canonicalized-url/127036/6 by Rob Biedenharn
```